Sources/AWSXRayInstrument/Instrument.swift

```diff
@@ -0,0 +1,40 @@
+import AWSXRayRecorder
+import Baggage
+import Dispatch // DispatchTime
+import Instrumentation
+import NIOHTTP1 // HTTPHeaders
+import NIOInstrumentation // HTTPHeadersExtractor, HTTPHeadersInjector
+
+extension XRayRecorder: TracingInstrument {
+    public func extract<Carrier, Extractor>(_ carrier: Carrier, into baggage: inout BaggageContext, using extractor: Extractor) where Carrier == Extractor.Carrier, Extractor: ExtractorProtocol {
+        guard
+            let headers = carrier as? HTTPHeaders,
```

Comment:
Not sure why a cast to `HTTPHeaders` and `HTTPHeadersExtractor` is necessary. That way you'll limit the use cases of your instrument to just HTTP headers.

Reply:
💯 

you are absolutely right, I think Xcode had some type resolving problems when I was writing it
removed casting and works now, plus it fixed the unit test :-)